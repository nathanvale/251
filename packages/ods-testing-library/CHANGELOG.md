# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
