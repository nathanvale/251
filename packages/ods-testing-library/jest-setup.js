import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
