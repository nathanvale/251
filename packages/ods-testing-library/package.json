{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
