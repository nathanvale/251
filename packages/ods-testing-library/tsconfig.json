{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./dist/typings",
    "rootDir": "./src"
  },
  "include": [
    "src",
    "src/**/*.json",
    "../../globals.d.ts",
    "../../@types",
    "jest-setup.ts"
  ],
  "references": [{"path": "../ods-themes/tsconfig.json"}]
}
