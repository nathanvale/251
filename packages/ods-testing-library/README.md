# @origin-digital/ods-testing-library

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Installation](#installation)
- [Why do I need this package?](#why-do-i-need-this-package)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

This package enhances the render function of `react-testing-library` so that the `<OriginThemeProvider/>` is wrapped around your rendered component.

All other API features of `react-testing-library` are also included.

## Installation

Install the package in your project directory with:

```sh
yarn add @origin-digital/ods-testing-library
```

This package also has a peer dependency on the following:

```sh
// with yarn
yarn add @origin-digital/ods-core
yarn add -D @testing-library/react
yarn add -D jest-styled-components
```

## Why do I need this package?

**The problem**

When using react-testing-library and ODS with the jest testing framework it is necessary to wrap **all** your components in our `<OriginThemeProvider/>` - otherwise the react component cannot be rendered and tested.

**The solution**

`@origin-digital/ods-testing-library` creates a wrapper around the `react-testing-library` render function. It is here that we wrap the `<OriginThemeProvider/>`. All other API features of `react-testing-library` are also included.

This practice is recommended by the `react-testing-library` documentation itself...

https://testing-library.com/docs/native-testing-library/setup#custom-render

The origin-ui dogfoods this package itself but it can also be used in your own app to test by adding to your jest config:

```
{
...,
setupFilesAfterEnv: [@origin-digital/ods-testing-library/jest-setup],
...
}
```

and usage in your tests:

```
import React from 'react';
import {render} from '@origin-digital/ods-testing-library';
import {Box} from '../Box/Box';
import {Stack} from './Stack';

test('It can stack', () => {
  const {container} = render(
    <Stack>
      <Box backgroundColor="red" padding="medium" />
      <Box backgroundColor="orange" padding="medium" />
      <Box backgroundColor="blue" padding="medium" />
    </Stack>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
```
