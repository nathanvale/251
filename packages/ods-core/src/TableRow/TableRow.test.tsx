import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Table, TableBody, TableRow, TableCell, TableRowProps } from "..";

const renderTableRow = (tableRowProps: Partial<TableRowProps> = {}) =>
  render(
    <Table>
      <TableBody>
        <TableRow {...tableRowProps}>
          <TableCell>Hello world</TableCell>
        </TableRow>
      </TableBody>
    </Table>
  );

test("It renders", () => {
  const { container } = renderTableRow();
  expect(container).toMatchSnapshot();
});
test("It adds class name", () => {
  const className = "myCustomClass";
  const { container, getAllByTestId } = renderTableRow({
    className,
  });
  expect(getAllByTestId("table-row").pop()?.classList).toContain(className);
  expect(container).toMatchSnapshot();
});
