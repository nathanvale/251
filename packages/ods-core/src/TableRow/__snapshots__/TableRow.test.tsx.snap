// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It adds class name 1`] = `
<div>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
      data-id="table"
    >
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root myCustomClass"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-22 TableCell-withPadding-27 TableCell-withVerticalAlign-21 TableCell-withVerticalAlign-26 TableCell-withFontSize-24 TableCell-withFontSize-28 TableCell-withoutFirstLastPadding-23"
            data-id="table-cell"
          >
            Hello world
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`It renders 1`] = `
<div>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
      data-id="table"
    >
      <tbody
        class="MuiTableBody-root"
        data-id="table-body"
      >
        <tr
          class="MuiTableRow-root"
          data-id="table-row"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body TableCell-withPadding-8 TableCell-withPadding-13 TableCell-withVerticalAlign-7 TableCell-withVerticalAlign-12 TableCell-withFontSize-10 TableCell-withFontSize-14 TableCell-withoutFirstLastPadding-9"
            data-id="table-cell"
          >
            Hello world
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
